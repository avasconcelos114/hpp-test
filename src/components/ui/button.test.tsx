import React from 'react';
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Button } from './button';

describe('Button', () => {
  it('should render the button', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText('Click me')).toBeDefined();
  });

  it('should have default variant and size classes', () => {
    render(<Button>Default</Button>);
    const btn = screen.getByRole('button');
    expect(btn.className).toContain('bg-bvnk-primary');
  });

  it('should have ghost variant class', () => {
    render(<Button variant='ghost'>Ghost</Button>);
    const btn = screen.getByRole('button');
    expect(btn.className).toContain('hover:bg-accent');
    expect(btn.className).toContain('hover:text-accent-foreground');
  });

  it('should have link variant class', () => {
    render(<Button variant='link'>Link</Button>);
    const btn = screen.getByRole('button');
    expect(btn.className).toContain('text-primary');
    expect(btn.className).toContain('hover:underline');
  });

  it('should have small size class', () => {
    render(<Button size='sm'>Small</Button>);
    const btn = screen.getByRole('button');
    expect(btn.className).toContain('h-8');
    expect(btn.className).toContain('px-3');
  });

  it('should have large size class', () => {
    render(<Button size='lg'>Large</Button>);
    const btn = screen.getByRole('button');
    expect(btn.className).toContain('h-10');
    expect(btn.className).toContain('px-6');
  });

  it('should have icon size class', () => {
    render(<Button size='icon'>Icon</Button>);
    const btn = screen.getByRole('button');
    expect(btn.className).toContain('size-9');
  });
});
