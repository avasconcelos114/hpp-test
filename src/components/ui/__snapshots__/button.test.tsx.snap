// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render the button 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-status-error/20 dark:aria-invalid:ring-status-error/40 aria-invalid:border-status-error cursor-pointer bg-bvnk-primary text-white shadow-xs hover:bg-bvnk-primary/90 h-[40px] px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
  data-testid="button"
>
  Click me
</button>
`;
